concepts/numbers/about.md

```diff
@@ -0,0 +1,160 @@
+# About Numbers
+
+## Numerical types
+
+[Numbers][ref-numbers] can be integer, unsigned integer, or floating point types.
+Each comes in various "sizes", meaning how many bits it needs in memory.
+
+Unlike some scripting languages (Ruby, recent versions of Python), each numeric type in Kotlin has a maximum ([`MAX_VALUE`][ref-max_value]) and minimum ([`MIN_VALUE`][ref-min_value]) value it can store.
+Assigning larger values will cause ["overflow"][wiki-overflow], resulting in either an exception (_bad_) or corrupted data (_worse_).
+
+```kotlin
+Int.MAX_VALUE  // => 2147483647 (maximum for a 32-bit signed integer)
+Int.MAX_VALUE + 1  // -2147483648 (overflow gives a negative result!)
+```
+
+- Integers can be `Byte`, `Short`, `Int` or `Long`, respectively 8, 16, 32 and 64 bits (1, 2 4, 8 bytes), and can be positive or negative.
+- Unsigned integers have a `U` prefix: `UByte`, `UShort`, `UInt` or `ULong`, and can only represent numbers ≥ 0.
+- Floating point types are `Float` (32-bit) or `Double` (64-bit).
+
+Integer variables relying on type inference default to `Int`, even on 64-bit machines, but floating point variables default to `Double`.
+
+Other types can, of course, be specified, but there are a few syntactic shortcuts, and big integer literals become `Long` if they would overflow `Int`.
+
+```Kotlin
+val one = 1 // defaults to Int
+val threeBillion = 3_000_000_000 // Long, with optional underscores for clarity
+val oneLong = 1L // Long
+val oneByte: Byte = 1
+val oneDouble = 1.0 // defaults to Double
+val oneFloat = 1.0f // Float
+val lightSpeed = 3.0e8 // scientific notation (units of m/s)
+```
+
+Hexadecimal and binary literals are the same as in many languages: `0x7F` and `0b100101` respectively.
+Octal literals are not supported in Kotlin.
+
```

Comment:
Maybe also add the Kotlin does not support arbitrary precision arithmetic in its standard library, so a userland solution is necessary for `BigInteger` and `BigDecimal` support?